tests/unit/nupic/research/spatial_pooler_unit_test.py

```diff
@@ -1786,6 +1788,57 @@ def testWrite(self):
     self.assertSetEqual(indices1, indices2)
 
 
+# moved from flat_spatial pooler tests: 
+
+  def testRandomSPDoesNotLearn(self):
+
+    sp = SpatialPooler(inputDimensions=[5],
+                       columnDimensions=[10])
+    inputArray = (numpy.random.rand(5) > 0.5).astype(uintDType)
+    activeArray = numpy.zeros(sp._numColumns).astype(realDType)
+    # Should start off at 0
+    self.assertEqual(sp._iterationNum, 0)
+    self.assertEqual(sp._iterationLearnNum, 0)
+
+    # Store the initialized state
+    initialPerms = copy(sp._permanences)
+
+    sp.compute(inputArray, False, activeArray)
+    # Should have incremented general counter but not learning counter
+    self.assertEqual(sp._iterationNum, 1)
+    self.assertEqual(sp._iterationLearnNum, 0)
+
+    # Check the initial perm state was not modified either
+    self.assertEqual(sp._permanences, initialPerms)
+
+
+  @unittest.skip("Ported from FlatSP and fails") #FIXME
```

Comment:
Why does this fail? If there's a good reason it doesn't make sense to fix this now, please update the skip comment and elaborate.